### ISSUE TITLE
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_1412","uid":"ipcp_1558588008096_498","cmd":"CS_ACTIVATE_ME","args":{}}]

### ISSUE THREAD

**Issue by khaninejad** [A9T9/RPA#74]
Hi,

There is an issue when Kantu enabled in Firefox browser, I see this errors on my console after 4-5 second of opening a page:
TypeError: e.data.indexOf is not a function[Learn More] post2.html:10:13
    post https://duckduckgo.com/post2.html:10
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588260650_438","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588260667_860","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588260701_662","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588258932_624","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588258949_144","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588259007_38","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588259020_930","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588259032_779","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588259050_355","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588259304_919","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588259320_906","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588259351_177","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588260599_555","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588260616_821","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588260634_994","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588260310_546","cmd":"CS_RECORD_ADD_COMMAND","args":{"target":"link=GitHub - A9T9/Kantu: Kantu for Chrome and Firefox - Modern Web Browser ...","targetOptions":["link=GitHub - A9T9/Kantu: Kantu for Chrome and Firefox - Modern Web Browser ...","xpath=//*[@id=\"r1-0\"]/div/h2/a[1]","css=#r1-0 > div > h2 > a.result__a"],"cmd":"click","url":"https://duckduckgo.com/?q=kantu+github&t=ffab&atb=v170-2__&ia=web"}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588260373_49","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588260406_57","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588260226_532","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588258633_193","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588255003_235","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
Promised response from onMessage listener went out of scope, sendMessage, [{"type":"CS_ASK_BG_4474","uid":"ipcp_1558588261015_753","cmd":"CS_ACTIVATE_ME","args":{}}] content_script.js:1:1908
TypeError: e.data.indexOf is not a function[Learn More] post2.html:10:13

​It's more like a sniffer !!

**Comment by A9T9** [comment]
Hi, I have problems recreating this issue. Can you please add a screenshot of the situation? Thanks.

**Comment by A9T9** [comment]
And this happens on any website?

**Comment by A9T9** [comment]
and if you test kantu for Chrome, do you see the same?

**Comment by khaninejad** [comment]
Yes, its happen on every website. but let me test on chrome.

**Comment by khaninejad** [comment]
this error occurred on chrome.
![error](https://user-images.githubusercontent.com/4404500/58245292-ac768f80-7d69-11e9-957d-da4b0edb31a5.PNG)

**Comment by A9T9** [comment]
Wow, I have never seen this error. Where you able to install kantu in chrome?

**Comment by khaninejad** [comment]
This my workstation, what do you mean by where?

**Comment by A9T9** [comment]
Typo ;-)

I meant: Were

=> Can you install Kantu for Chrome at all?

**Comment by khaninejad** [comment]
now suddenly everything works perfectly!

**Comment by metinc** [comment]
Hi, I'm seeing the same bug in Firefox and it seems to be tricky to recreate it. The most reliable way I found to recreate it is this:

1. If you have Firefox running then close it.
2. Open Firefox.
3. Open the Dev Tools (ctrl + shift + i)
4. Enter https://stackoverflow.com/ into your address bar and load it.
5. While it is loading keep clicking on the speech bubble icon in the upper right multiple times.
![lemmesmash](https://user-images.githubusercontent.com/1820121/59094471-7c58f000-8916-11e9-958c-4f1c6f63afa6.png)


After 5 to 10 seconds the error should appear. I can not recreate this after refreshing the page and I have to close Firefox first.
Tested with Firefox 67.0.1 and Kantu 5.0.1
Also I noticed that Firefox is slowing down after some time when I have Kantu Add-On enabled. Maybe there is a memory leak.
